opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/QueryTest.java

```diff
@@ -0,0 +1,132 @@
+        dsl.namedArgument("boost", DSL.literal("1")),
+        dsl.namedArgument("default_operator", DSL.literal("AND")),
+        dsl.namedArgument("default_operator", DSL.literal("and")),
+        dsl.namedArgument("enable_position_increments", DSL.literal("true")),
+        dsl.namedArgument("escape", DSL.literal("false")),
+        dsl.namedArgument("fuzziness", DSL.literal("1")),
+        dsl.namedArgument("fuzzy_rewrite", DSL.literal("constant_score")),
+        dsl.namedArgument("fuzzy_max_expansions", DSL.literal("42")),
+        dsl.namedArgument("fuzzy_prefix_length", DSL.literal("42")),
+        dsl.namedArgument("fuzzy_transpositions", DSL.literal("true")),
+        dsl.namedArgument("lenient", DSL.literal("true")),
+        dsl.namedArgument("max_determinized_states", DSL.literal("10000")),
+        dsl.namedArgument("minimum_should_match", DSL.literal("4")),
+        dsl.namedArgument("quote_analyzer", DSL.literal("standard")),
+        dsl.namedArgument("phrase_slop", DSL.literal("0")),
+        dsl.namedArgument("quote_field_suffix", DSL.literal(".exact")),
+        dsl.namedArgument("rewrite", DSL.literal("constant_score")),
+        dsl.namedArgument("type", DSL.literal("best_fields")),
+        dsl.namedArgument("tie_breaker", DSL.literal("0.3")),
+        dsl.namedArgument("time_zone", DSL.literal("Canada/Pacific")),
+        dsl.namedArgument("ANALYZER", DSL.literal("standard")),
+        dsl.namedArgument("ANALYZE_wildcard", DSL.literal("true")),
+        dsl.namedArgument("Allow_Leading_wildcard", DSL.literal("true")),
+        dsl.namedArgument("Auto_Generate_Synonyms_Phrase_Query", DSL.literal("true")),
+        dsl.namedArgument("Boost", DSL.literal("1"))
+    ).stream().map(arg -> List.of(query, arg));
+  }
+
+  @ParameterizedTest
+  @MethodSource("generateValidData")
+  public void test_valid_parameters(List<Expression> validArgs) {
+    Assertions.assertNotNull(queryQuery.build(
+        new QueryExpression(validArgs)));
+  }
+
+  @Test
+  public void test_SyntaxCheckException_when_no_arguments() {
+    List<Expression> arguments = List.of();
+    assertThrows(SyntaxCheckException.class,
+        () -> queryQuery.build(new QueryExpression(arguments)));
+  }
+
+  @Test
+  public void test_SyntaxCheckException_when_field_argument() {
+    List<Expression> arguments = List.of(
+        namedArgument("fields", "invalid argument"),
+        namedArgument("query", query_value));
+    assertThrows(SemanticCheckException.class,
+        () -> queryQuery.build(new QueryExpression(arguments)));
+  }
+
+  @Test
+  public void test_SemanticCheckException_when_invalid_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("query", query_value),
+        namedArgument("unsupported", "unsupported_value"));
+    Assertions.assertThrows(SemanticCheckException.class,
+        () -> queryQuery.build(new QueryExpression(arguments)));
+  }
+
```

Comment:
add a test that sends the same namedArgument twice for code coverage

Reply:
Fixed in f2d77c634b1869229cbb3aa2f6952403d26fe30c